Update epoxy upgrade doc (#1857)
stackhpc/stackhpc-kayobe-config: doc/source/operations/upgrading-openstack.rst

      kayobe kolla ansible run "rabbitmq-upgrade 4.1"

.. _python-3-12:

Python 3.12
-----------

From OpenStack 2025.1, Kayobe and Kolla-Ansible require Python 3.12.

> Ubuntu 24.04 has default Python of version 3.12.

Reviewer: ```suggestion
Ubuntu 24.04 has a default Python of version 3.12.
```

You can find the upgrade procedure from :ref:`upgrading-to-ubuntu-noble`

For Rocky Linux 9, install Python 3.12 manually.

.. code-block:: bash

   dnf install python3.12

For both Operating Systems, Kayobe and Kolla-Ansible Python virtual environments

> created with older Python version will not work with OpenStack 2025.1.

Reviewer: ```suggestion
created with older Python versions will not work with OpenStack 2025.1.
```

Create new Kayobe environment and bootstrap Ansible control host with Python 3.12.
Beokay is recommended when creating and managing local Kayobe environment.

> You can find more information from :ref:`beokay` documentation.

Reviewer: ```suggestion
Create a new Kayobe environment and bootstrap the Ansible control host with Python 3.12.
Beokay is recommended when creating and managing the local Kayobe environment.
You can find more information from the :ref:`beokay` documentation.
```

.. note::

>    For Rocky Linux 9, ``beokay create`` need ``--python python3.12`` option

Reviewer: ```suggestion
   For Rocky Linux 9, ``beokay create`` must be used with the ``--python python3.12`` option
```

      kayobe kolla ansible run "rabbitmq-upgrade 4.1"

Python 3.12
-----------

> From OpenStack 2025.1, Kayobe and Kolla-Ansible requires Python 3.12.

Reviewer: ```suggestion
From OpenStack 2025.1, Kayobe and Kolla-Ansible require Python 3.12.
```

Author: Done

Ubuntu 24.04 has default Python of version 3.12.
You can find the upgrade procedure from :ref:`upgrading-to-ubuntu-noble`

For Rocky Linux 9, install Python 3.12 manually.

.. code-block:: bash

>    dnf install python3.12

Reviewer 2: Don't we need python3.12-devel?

Author: I can't find aio or multinode setup script/playbooks installing python3.12-devel for RL9. So probably not?

Upgrading local Kayobe environment
----------------------------------

.. note::

   For OpenStack 2025.1 Kayobe environment, Python 3.12 is required.
   You can find more information at :ref:`python-3-12`

> 

Reviewer: In this release, we _can't_ upgrade the same env. We should make that very clear

```suggestion
.. warning::

   Python 3.12 is required for OpenStack 2025.1 Kayobe environments.
   The environment cannot be upgraded for this release, it must be rebuilt.
   You can find more information at :ref:`python-3-12`

```

Upgrading local Kayobe environment
----------------------------------

.. warning::

   Python 3.12 is required for OpenStack 2025.1 Kayobe environments.
   The environment cannot be upgraded for this release, it must be rebuilt.
   You can find more information at :ref:`python-3-12`

The local Kayobe environment should be either recreated or upgraded to use the
new release. It may be beneficial to keep a Kayobe environment for the old
release in case it is necessary before the upgrade begins.

In general it is safer to rebuild an environment than upgrade. You can follow

> instructions from :ref:`beokay` documentation.

Reviewer: ```suggestion
instructions from the :ref:`beokay` documentation.
```
